### DISCUSSION TITLE
Unite Dialer and Contacts into one app?

### DISCUSSION THREAD

**Discussion by inson1** [FossifyOrg/General-Discussion#564] [Polls]
I think it would be easier even for developers

For users Why have 2 apps?

PS Please comment your reason Why not Why yes bellow.

**Comment by GitCurious** [comment]
Already commented on the other poll but I think there's so much crossover between the 2 apps, they could easily be just one.

So I voted 100% Yes !

**Comment by Aga-C** [comment]
Dialer used to be a part of Contacts. Here is why it was separated: https://github.com/SimpleMobileTools/Simple-Contacts/issues/224#issuecomment-451725248.

**Reply by inson1** [reply]
Which is my biggest reason Why I hate it, without call history contacts are useless and you have have 2 apps on home screen. If that info was in contacts I could just open contacts and do everything from them

**Reply by sphh** [reply]
Wouldn't the same argument go for without text message history contacts are useless. Or email history? Or chat history? (Put your preferred way of communication channel history here). Unless we get an app, which connects contacts with phone calls, text messages, emails, chats and messenger history, we have to live with separate apps, which do one thing and do it well.

**Reply by inson1** [reply]
@sphh If contacts had also sms history that would be great. And if it also had email, chat histories etc. that would be super power. 
But because it should be simple and mainly for phone numbers, then email and chat histories shouldnt be there.

**Comment by tswistak** [comment]
I don't use Simple Contacts, but I use Simple Dialer. I prefer my stock Contacts app because it's IMHO prettier (I blocked Internet access from it anyway) and the main reason I use Dialer is that the proximity sensor in my phone is broken, and it's the only dialer where I could turn it off.

**Comment by naveensingh** [comment]
@inson1 I said it's not a bad idea but I didn't say we will do it now. I'm keeping this open but we won't be unifying anything anytime soon regardless of the poll results.

**Reply by inson1** [reply]
I didnt say we should do that now, I just created discussion so we would know what community wants and reasons why its good and bad idea

**Comment by inson1** [comment]
https://www.reddit.com/r/Fossify/comments/19190zi/unite_dialer_and_contacts_into_one_app/

**Comment by siSSSou** [comment]
Please Don't !
Contacts are not only used to make phone calls.
It's useful to have an app for contacts ans another just for phone.

**Reply by DoctorSubtilis** [reply]
I have to use contacts to phone, because only in contacts I can use **groups**.

**Reply by Saarsk** [reply]
This is a good argument against. I have in the past used only the contacts app but another for dialer. If one might want to or need to change the dialer app, you'd lose both in the prosess if they were to be merged.

I wonder if there could be a "both" solution, perhaps if the apps were structured as modules that can reside inside the same "app icon" (from the user's perspective?). Perhaps kind of like a core app, that (depending on the user preference) have either the dialer module or the contacts module. Or both.

**Comment by TheSmilingTurtle** [comment]
I think the two apps should be united into one aswell, as there is a lot of overlap.

**Comment by Mr-Bajs** [comment]
I would prefer not to. I find that the contacts app useful but i've had problems with the phone app before. Giving the user the choice to the apps that they prefer and not having to "buy" into the suite is a feature as I see it an not the other way around. I consider apps that I don't use as bloat and prefer not having apps i don't use on the phone. 

Also in terms of security and privacy isn't it better that the apps are seperate and using the android build in sandboxing of the apps. As an end user it is much easier to manage and understand privilages to apps that have a clear usage.

Wouldn't it also be easier to maintain if the apps where seperate?

**Reply by laundmo** [reply]
I don't see why uniting the apps would neccesitate using both features. You could still select a different dialer in the contacts app, and disable the UI elements.

**Comment by DoctorSubtilis** [comment]
I think that a **group** tab is very useful to do a call, if you have many contacts, and **group** tab is only in Contacts.

**Comment by sphh** [comment]
I also prefer to have two different apps, so that I can use the Fossify Phone app and another contacts app (or the other way round)! Please leave us that choice.

Personally I do not see the overlap mentioned. For me the only similarity between the Phone and Contacts app is the list of contacts. Mind you, I do not use the Phone app to edit contacts, so I might miss a lot here.

**Reply by Saarsk** [reply]
There is overlap for sure. But I still think there are good reasons for keeping them separate. Though, I'd prefer the UX to be (optionally) merged without actually merging the apps. That way we'd have both.

**Comment by DoctorSubtilis** [comment]
The only reason to unify contacts and dialer is, in my opinion, the possibility to use **groups** in dialer, without have to go out and back every time.   

Therefore either:
- unify contacts and dialer
- **or** add **groups** to dialer  

But nobody here is interested in using groups?

**Reply by sphh** [reply]
@DoctorSubtilis: Have you opened a feature request to include group functionality into the dialer?

**Reply by DoctorSubtilis** [reply]
Yes, I have.

**Reply by sphh** [reply]
Could you please add a link here for reference. Thanks.

**Reply by DoctorSubtilis** [reply]
Sorry, I didn't manage to find it (archivied?): I remember only that I was said that it was not possible to add groups to dialer.

**Reply by inson1** [reply]
@DoctorSubtilis @sphh https://github.com/FossifyOrg/Phone/issues/75

**Reply by Saarsk** [reply]
Groups would indeed be useful. You could access them from the contacts app, but that would require a separate icon and you'd have to configure contacts to call when pressing on a contact which I personally wouldn't want. And you'd have to remember to not use the dialer for certain situation. It would be kinda messy.

So yeah, groups in the dialer makes total sense.

**Comment by Saarsk** [comment]
@naveensingh 
How viable would it be to structure the app in such way that it would solve for both sides? Specifically, would it be an idea to have a core app that could be expanded (populated with) both a dialer module and a contacts module, or just one of them?

I voted for merging, but I regret my vote and now I can't change it. I forgot the many reasons why having them separate can be very beneficial. But merging can also be beneficial. So, could there be both, without actually needing two separate app projects?

The structure of the app could be unified, but the population of elements, tabs, menus and additions would depend on the modules (kind of like plugins).

So without any modules (not possible, but for the sake of describing my idea), the core alone would simply be an app that could be populated with pages/tabs, settings, toolbar(s), dialog options, list entry structure(s) and different states and menus/operations. Functionality-wise, there's a lot of overlap. But use-case-wise, it depends on different users, preferences and situations.

I wonder if this structure, the elements, the states, settings and variables could be requested/created by either module. They could add e.g. a new page or merge (depending on whether the user choose to merge into one app or separate into two. Or, just have one of the modules, which in terms of code would be kind of the same as separate mode only there's just one of the modules present).

I think it's fairly easy to enable or disable additional app icons as well. But I guess the question is, does the OS require two separate APKs in order to set the default for both dialer and contacts? Or could a single APK be both? If the latter, this would solve both these sides but also add new advantages. I find this idea kind of intriguing.

I realize it would require a (partial?) rewrite, but what if other apps could benefit from a core + modules approach as well? Like e.g. File Manager and Gallery. And on some level, perhaps all apps could reuse resources/dependencies to keep the app size(s) low.

Maybe it would be easier to change menus, buttons and functions on all apps by simply changing a global parameter. Since multiple apps have a search bar, the same toolbars, sometimes the same menus and operations.

If it was modular, any app could acquire a global module, element, toolbar, menu or funtion. But also deviate from the global parameters by explicitly add exceptions or additions. If all the components of the apps are modules and parameters, wouldn't it solve a lot of development problems? I imagine it would save a lot of time and energy in the long-term. And also make the apps extremely flexible in terms of functionality, as well as solid due to the core elements being made once and with modularity by design.

(I believe the suite shares libraries and components today, but I don't know to which extent and how much flexibility and scalability it enables.)

**Reply by naveensingh** [reply]
I have thought about that 2 In 1 idea but I'm still not convinced that the apps should be merged. They work fine independently.

> I believe the suite shares libraries and components today, but I don't know to which extent and how much flexibility and scalability it enables.

It was always 'simple'. Any function/string/resource that is required by more than one app is directly added to the commons library and all Fossify apps depend on the commons library.

> I voted for merging, but I regret my vote and now I can't change it.

You can. Click on 'Hide results' => Select your preferred option and click vote.

**Comment by DoctorSubtilis** [comment]
But what make impossible to keep a contacts app and to show groups _also_ in dialer one?
I mean: contacts are shown both in contacts and in dialer. 
Why it wouldn't be possible have also groups, **both** in contacts **and** in dialer?

**Comment by naveensingh** [comment]
Closing because they won't be merged, they are different enough even if they look similar UI-wise.

**Comment by NickAndreou** [comment]
Never even cared to download this Contacts app - I have everything I need in the Phone app.

**Reply by naveensingh** [reply]
> I have everything I need in the Phone app

How do you edit your contacts? You must have a contacts app installed because Fossify Phone doesn't do editing.

**Reply by NickAndreou** [reply]
> > I have everything I need in the Phone app
> 
> How do you edit your contacts? You must have a contacts app installed because Fossify Phone doesn't do editing.

I never had a problem editing contacts in your Phone app. Why do you say it can't be done?
